### PR TITLE
[FIX] mail: only active users could be used for chat window creation

### PR DESCRIPTION
**Pull request by xdu-odoo**
task-2263679

### CODE REVIEW

**Review by alexkuhn — CHANGES_REQUESTED** [review]
Runbot red (im livechat typing notification test fails)

File: addons/im_livechat/static/src/messaging/component/thread_icon/thread_icon_tests.js [hunk 1]
```diff
@@ -68,7 +68,11 @@ QUnit.test('livechat: public website visitor is typing', async function (assert)
                 }],
             }],
         },
-        public_partner: [7, "Public Partner"],
+        public_partner: {
+            active: false,
+            id:7,
+            display_name: "Public Partner",
+        },
```
Comment (seb-odoo) [inline addons/im_livechat/static/src/messaging/component/thread_icon/thread_icon_tests.js]: alphabetical, space after id:

File: addons/mail/models/res_partner.py [hunk 2]
```diff
@@ -54,6 +54,13 @@ def find_or_create(self, email, assert_valid_email=False):
 
         return super(Partner, self).find_or_create(email, assert_valid_email=assert_valid_email)
 
+    def mail_partner_format(self):
+        return {
```
Comment (seb-odoo) [inline addons/mail/models/res_partner.py:59]: add self.ensure_one

File: addons/im_livechat/static/src/messaging/component/thread_textual_typing_status/thread_textual_typing_status_tests.js [hunk 3]
```diff
@@ -73,7 +73,11 @@ QUnit.test('receive visitor typing status "is typing"', async function (assert)
                 }],
             }],
         },
-        public_partner: [7, "Public Partner"],
+        public_partner: {
+            active: false,
+            id:7,
+            display_name: "Public Partner",
+        },
```
Comment (seb-odoo) [inline addons/im_livechat/static/src/messaging/component/thread_textual_typing_status/thread_textual_typing_status_tests.js]: same

File: addons/mail/static/src/messaging/test_utils.js [hunk 4]
```diff
@@ -456,8 +456,16 @@ function beforeEach(self) {
...
+            partner_root: {
+                active: false,
+                id:2,
+                display_name: "OdooBot",
+            },
+            public_partner: {
+                active: false,
+                id:4,
+                display_name: "Public user",
+            },
```
Comment (seb-odoo) [inline addons/mail/static/src/messaging/test_utils.js]: same

File: addons/mail/static/tests/helpers/mock_server.js [hunk 5]
```diff
@@ -143,16 +143,24 @@ MockServer.include({
...
+            partner_root: {
+                active: false,
+                id:2,
+                display_name: "OdooBot",
+            },
+            public_partner: {
+                active: false,
+                id:4,
+                display_name: "Public user",
+            },
```
Comment (seb-odoo) [inline addons/mail/static/tests/helpers/mock_server.js]: same

File: addons/im_livechat/static/src/messaging/component/thread_textual_typing_status/thread_textual_typing_status_tests.js [hunk 6]
```diff
@@ -73,7 +73,11 @@ QUnit.test('receive visitor typing status "is typing"', async function (assert)
                 }],
             }],
         },
-        public_partner: [7, "Public Partner"],
+        public_partner: {
+            active: false,
+            display_name: "Public Partner",
+            id :7,
```
Comment (seb-odoo) [inline addons/im_livechat/static/src/messaging/component/thread_textual_typing_status/thread_textual_typing_status_tests.js]: space after : not before

File: addons/mail/static/tests/helpers/mock_server.js [hunk 7]
```diff
@@ -143,16 +143,24 @@ MockServer.include({
...
-            'starred_counter': 0,
+            channel_slots: [],
+            commands: [],
+            mail_failures: [],
+            mention_partner_suggestions: [],
+            menu_id: false,
+            needaction_inbox_counter: 0,
+            partner_root: {
+                active: false,
+                id:2,
```
Comment (seb-odoo) [inline addons/mail/static/tests/helpers/mock_server.js]: space

**Review by seb-odoo — APPROVED** [review]
I didn't test but code looks ok.

We'll probably have to add "active" in more formatters to cover more cases (actual archived partners in message fetch, etc) but this can be done later/when the need arises.

cc @alexkuhn you can check this

**Review by alexkuhn — APPROVED** [review]

**Pull request merged**